integration/spec/support/email_output_service/email_attachment_extractor.rb

```diff
@@ -1,16 +1,17 @@
 class EmailAttachmentExtractor
-  def self.find(id:, pdf_filename:, user_attachment_filename:)
-    email_finder = EmailFinder.new(
-      service: GoogleService,
-      id: id,
-      pdf_filename: pdf_filename,
-      user_attachment_filename: user_attachment_filename
-    )
-
+  def self.find(
+    id:,
+    expected_emails: nil,
+    find_criteria: nil
+  )
     tries = 1
     max_tries = 20
 
     until tries > max_tries
+      email_finder = EmailFinder.new(id: id, service: GoogleService)
```

Comment:
On every try, instantiate the object again so I can cache the calls to the gmail by instance (every try).